osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/misc/GetGsmDiagnosticCommandExecutorTest.java

```diff
@@ -0,0 +1,263 @@
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.commands.utils.DlmsHelper;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.DlmsDevice;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.entities.Protocol;
+import org.opensmartgridplatform.adapter.protocol.dlms.domain.factories.DlmsConnectionManager;
+import org.opensmartgridplatform.adapter.protocol.dlms.exceptions.ProtocolAdapterException;
+import org.opensmartgridplatform.adapter.protocol.dlms.infra.messaging.DlmsMessageListener;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.BitErrorRateDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.CircuitSwitchedStatusDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.GetGsmDiagnosticRequestDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.GetGsmDiagnosticResponseDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.ModemRegistrationStatusDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.PacketSwitchedStatusDto;
+import org.opensmartgridplatform.dto.valueobjects.smartmetering.SignalQualityDto;
+
+@ExtendWith(MockitoExtension.class)
+@MockitoSettings(strictness = Strictness.LENIENT)
+public class GetGsmDiagnosticCommandExecutorTest {
+
+  @InjectMocks private GetGsmDiagnosticCommandExecutor executor;
+
+  @Mock private DlmsMessageListener dlmsMessageListener;
+
+  @Mock private DlmsHelper dlmsHelper;
+
+  @Mock private DlmsObjectConfigService dlmsObjectConfigService;
+
+  @Mock private DlmsConnectionManager connectionManager;
+
+  private final DlmsDevice device = this.createDevice(Protocol.SMR_5_1, CommunicationMethod.CDMA);
+  private final long from = 1111111L;
+  private final long to = 2222222L;
+  private final int classId = 99;
+  private final String obiscode = "1.2.3.4.5.6";
+  private final GetGsmDiagnosticRequestDto request = new GetGsmDiagnosticRequestDto();
+
+  @BeforeEach
+  public void setUp() {
+    when(this.connectionManager.getDlmsMessageListener()).thenReturn(this.dlmsMessageListener);
+  }
+
+  @Test
+  public void testExecuteObjectNotFound() throws ProtocolAdapterException {
+    // SETUP
+    when(this.dlmsObjectConfigService.findDlmsObjectForCommunicationMethod(any(), any()))
+        .thenThrow(new ProtocolAdapterException("Object not found"));
+
+    // CALL
+    try {
+      this.executor.execute(this.connectionManager, this.device, this.request);
+      fail("When no matching object is found, then execute should fail");
+    } catch (final ProtocolAdapterException e) {
+      assertThat(e.getMessage()).isEqualTo("Object not found");
+    }
+  }
+
+  @Test
+  public void testBundle() throws ProtocolAdapterException {
+    // CALL
+    final GetGsmDiagnosticRequestDto request = this.executor.fromBundleRequestInput(this.request);
+
+    // VERIFY
+    // Verify that the executor creates a new request
+    assertThat(request).isNotEqualTo(this.request);
```

Comment:
Not sure if this is a very good check. If I were to implement an equals method on `GetGsmDiagnosticRequestDto` this would likely fail.
If you would like to check it is not the same object you could use `isNotSameAs`, but I'm not sure what would be useful about this. Given that the request looks like it is defined by its type and does not hold relevant data, what would be the problem if the same object was reused?